### PR TITLE
[JBS-47] Adding metrics to 1) Cinder Task Flows 2) DB 3) DSS Connection 4) Sch…

### PR DESCRIPTION
**Pull request by souvik-ray**
This pull request is continuation of 
https://github.com/JioCloud/cinder/pull/64/files

### CODE REVIEW

File: cinder/backup/drivers/sbs.py [hunk 1]
```diff
@@ -637,7 +644,7 @@ def _backup_rbd(self, backup, volume_file, volume):
                 raise exception.BackupOperationError(msg)
 
         #make sure snap is newer than base
-        now = dt.datetime.now()
```
Comment (vivekdhayaal) [inline cinder/backup/drivers/sbs.py]: Please restore the original line as this is the fix done by Shishir for backup deletion issue and so we need it.

**Comment by vivekdhayaal** [comment]
Two things:
1)Specify the JIRA bug ID in the commit message
2)Please share the test cases you ran and their respective outputs inline with the new merge process proposal by Shishir

**Comment by souvik-ray** [comment]
==== Create Volume successful ==== 

cinder create 1

Describe Volume successful
ubuntu@compute3:/var/log/cinder$ cinder list | grep d42b01d3-ffde-480c-aa9b-550362863235
| d42b01d3-ffde-480c-aa9b-550362863235 | available |                          None                         |  1   |     None    |  false   |             |


=== Cinder create backup from volume ===
+-----------+--------------------------------------+
|  Property |                Value                 |
+-----------+--------------------------------------+
|     id    | b5aa3a69-497e-4997-a465-a0c5261b9024 |
|    name   |                 None                 |
| volume_id | d42b01d3-ffde-480c-aa9b-550362863235 |
+-----------+--------------------------------------+


ubuntu@compute3:/var/log/cinder$ cinder backup-list | grep d42b01d3-ffde-480c-aa9b-550362863235
| b5aa3a69-497e-4997-a465-a0c5261b9024 | d42b01d3-ffde-480c-aa9b-550362863235 | available |                           None                          |  1   |     None     | sbs-devtest-backups-container |
| cb66fbdd-002c-484f-9b86-9d9bff419b12 | d42b01d3-ffde-480c-aa9b-550362863235 | available | volume-cb66fbdd-002c-484f-9b86-9d9bff419b12.backup.base |  2   |     None     | sbs-devtest-backups-container |

==== Backup Restore Failed (Snapshot Not found error. Does not seem to be related.) ====
ubuntu@compute3:/var/log/cinder$ cinder backup-restore d42b01d3-ffde-480c-aa9b-550362863235
ERROR: Not Found (HTTP 404) (Request-ID: req-34c480d7-1418-45a6-b345-077407791c39)
ubuntu@compute3:/var/log/cinder$ cinder backup-restore d42b01d3-ffde-480c-aa9b-550362863235
ERROR: Not Found (HTTP 404) (Request-ID: req-d665745b-89e8-444f-ad6f-997fccab083c)
ubuntu@compute3:/var/log/cinder$ cinder backup-restore d42b01d3-ffde-480c-aa9b-550362863235
ERROR: Not Found (HTTP 404) (Request-ID: req-2d560b73-aab9-4a04-ac6d-3e21bde6d279)

ubuntu@compute3:/var/log/cinder$ cinder backup-create d42b01d3-ffde-480c-aa9b-550362863235    (Backup delete)                                                                                           +-----------+--------------------------------------+
|  Property |                Value                 |
+-----------+--------------------------------------+
|     id    | 25025992-8026-4741-8117-d2c27c269630 |
|    name   |                 None                 |
| volume_id | d42b01d3-ffde-480c-aa9b-550362863235 |
+-----------+--------------------------------------+


==== Describe Backup ====
ubuntu@compute3:/var/log/cinder$ cinder backup-list | grep 25025992-8026-4741-8117-d2c27c269630
| 25025992-8026-4741-8117-d2c27c269630 | d42b01d3-ffde-480c-aa9b-550362863235 | available |                           None                          |  1   |     None     | sbs-devtest-backups-container |

==== Delete Backup =====
ubuntu@compute3:/var/log/cinder$ cinder backup-delete 25025992-8026-4741-8117-d2c27c269630
ubuntu@compute3:/var/log/cinder$ cinder backup-list | grep 25025992-8026-4741-8117-d2c27c269630
| 25025992-8026-4741-8117-d2c27c269630 | d42b01d3-ffde-480c-aa9b-550362863235 |  deleting |                           None                          |  1   |     None     | sbs-devtest-backups-container 

===== Delete Volume ====
ubuntu@compute3:/var/log/cinder$ cinder delete d42b01d3-ffde-480c-aa9b-550362863235
ubuntu@compute3:/var/log/cinder$ cinder list | grep d42b01d3-ffde-480c-aa9b-550362863235

**Comment by souvik-ray** [comment]
1) Added test cases.
2) Cinder create volume from backup is failing. It cannot find snapshots that are created. Does not look like changes introduced by me.
3) Could not test sqlalchemy db changes because the devtest is not compatible with current changes in the commit. That is the only file not tested but the annotations (in that file) are tested in other parts of the code multiple times.

**Comment by souvik-ray** [comment]
Output

==================================================================
StartTime=1460356958.46
EndTime=2016-04-11 12:12:45
Time=7070 ms
OperationName=cinder-backup-create
HostName=compute3
PID=1704
TenantId=82deb70d37c14d44a25b518e829802ca
RequestId=req-b1cf8532-62cf-4268-8b8f-96e72bf9bfeb
EOE
===================================================================
StartTime=1460348061.0
EndTime=2016-04-11 09:44:21
Time=64 ms
OperationName=cinder-backup-delete
HostName=compute3
PID=1704
TenantId=None
RequestId=req-db18e413-7035-40d2-8889-dfadf061e97a
EOE
===================================================================
StartTime=1460356992.1
EndTime=2016-04-11 12:13:14
Time=1989 ms
OperationName=cinder-backup-restore
HostName=compute3
PID=1704
TenantId=82deb70d37c14d44a25b518e829802ca
RequestId=req-a22ab347-6856-4517-8f41-616fa26d3241
EOE
===================================================================
StartTime=1460348099.03
EndTime=2016-04-11 09:44:59
Time=94 ms
OperationName=cinder-scheduler-volume-create
HostName=compute3
PID=59950
TenantId=82deb70d37c14d44a25b518e829802ca
RequestId=req-1e764948-8a99-4e82-ad16-49d53afeb082
EOE
===================================================================
StartTime=1460351603.46
EndTime=2016-04-11 10:43:25
Time=2452 ms
OperationName=cinder-volume-create
HostName=compute3
PID=14619
TenantId=82deb70d37c14d44a25b518e829802ca
RequestId=req-7f25da4b-84d8-44f4-a49b-0e9aef8c57b1
EOE
==================================================================
Scheduler



==================================================================
StartTime=1460364590.73
EndTime=2016-04-11 14:19:50
Time=108 ms
OperationName=cinder-scheduler-volume-create
HostName=compute3
PID=59950
TenantId=82deb70d37c14d44a25b518e829802ca
RequestId=req-c5b87d87-63ac-4831-95dd-b0839e7e0c31
ProgramName=CinderScheduler
MarketplaceId=IDC1
Timing=volume-rpcapi-create-volume_time=19 ms,ScheduleCreateVolumeTask_execute-time=83 ms,ExtractSchedulerSpecTask_execute-time=0 ms,
Counters=ExtractSchedulerSpecTask_execute-occurence=1,Fault=0,Success=1,ScheduleCreateVolumeTask_execute-occurence=1,Error=0,
EOE
===================================================================
StartTime=1460364590.77
EndTime=2016-04-11 14:19:50
Time=106 ms
OperationName=show
RemoteAddress=10.140.12.217
PathInfo=/82deb70d37c14d44a25b518e829802ca/volumes/b1ad1b21-2b3a-40d4-84cc-b227eb3263bb
HostName=compute3
PID=60055
--
EOE
===================================================================
StartTime=1460365052.78
EndTime=2016-04-11 14:27:32
Time=102 ms
OperationName=cinder-scheduler-volume-create
HostName=compute3
PID=59950
TenantId=82deb70d37c14d44a25b518e829802ca
RequestId=req-a9fb4a1f-c14a-448e-a45d-2167dbe3db41
ProgramName=CinderScheduler
MarketplaceId=IDC1
Timing=volume-rpcapi-create-volume_time=16 ms,ScheduleCreateVolumeTask_execute-time=75 ms,ExtractSchedulerSpecTask_execute-time=0 ms,
Counters=ExtractSchedulerSpecTask_execute-occurence=1,Fault=0,Success=1,ScheduleCreateVolumeTask_execute-occurence=1,Error=0,
EOE
===================================================================
StartTime=1460365052.89
EndTime=2016-04-11 14:27:33
Time=244 ms
OperationName=cinder-volume-create
HostName=compute3
PID=14619
TenantId=82deb70d37c14d44a25b518e829802ca
RequestId=req-a9fb4a1f-c14a-448e-a45d-2167dbe3db41

**Comment by vivekdhayaal** [comment]
Reviewed the code; Looks good to me.

Chirag, please review the test-cases and execution.

**Comment by aggarwalchirag** [comment]
Looks fine

**Pull request merged**